File: pset6/readability/readability.py
from cs50 import get_string

# Retrieves text from CLI
text = get_string('Text: ')

# Initalizes counters for L = letter, S = sentence, and W = word
L = 0
S = 0
W = 1

# Loops through every element of the text
for el in text:
    # Checks if alphabetical, if so adds 1 to L
    if el.isalpha():
        L += 1

    # If el is a space, it indicates a word. W is initialized as 1 because of the last word
    elif el == ' ':
        W += 1

    # Checks for the declared punctuation, which indicates a sentence
    elif el == '.' or el == '?' or el == '!':
        S += 1

# Calculates indices
L = (L / W) * 100
S = (S / W) * 100
index = 0.0588 * L - 0.296 * S - 15.8

# Handles special index values
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// Prototype of helper function
void count(int counter[], string text);

int main(void)
{
    string text = get_string("Text: ");

    // Initializes array that will be filled with letter, word, and sentence count
    int counter[3];
    count(counter, text);

    // Calculates L, S, and index
    float L = (counter[0] / (float) counter[1]) * 100.0;
    float S = (counter[2] / (float) counter[1]) * 100.0;

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);


    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
};

void count(int counter[], string text)
{
    // Initializes counters
    int nletters = 0;
    int nwords = 1; // Since the last word will not be proceeded by a space
    int nsentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
       if (isalpha(text[i]))
       {
           nletters++;
       }
       if (text[i] == ' ')
       {
           nwords++;
       }
       if (text[i] == '.' || text[i] == '!' || text[i] == '?')
       {
           nsentences++;
       }
    }
    // Enters the counters into the array
    counter[0] = nletters;
    counter[1] = nwords;
    counter[2] = nsentences;

};